lib/bundler_api/gem_info.rb

```diff
@@ -2,7 +2,7 @@
 
 # Return data about all the gems: all gem names, all versions of all gems, all dependencies for all versions of a gem
 class BundlerApi::GemInfo
-  DepKey = Struct.new(:name, :number, :platform)
+  DepKey = Struct.new(:name, :number, :platform, :required_ruby_version, :rubygems_version)
```

Comment:
I think we probably need two different classes—the old DepKey is for the current dependency API, and the new structure should be used only for the /info files. We don't want to query for the ruby and rubygems version and checksum in the regular dependency API, which also uses this class.